Credencials issues in npm publish task

The publish task is failing saying the token doesn't have permission. We should

1) Generate a new token
2) Set the registry url explicitly to match the [github actions example](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#publish-to-npmjs-and-gpr-with-npm)


https://github.com/lancedb/lancedb/actions/runs/5580304414/jobs/10197719342

```
Run mv */*.tgz .
npm notice 
npm notice 📦  vectordb@0.1.14
npm notice === Tarball Contents === 
npm notice 309B   .eslintrc.js                                
npm notice 1.1kB  CHANGELOG.md                                
npm notice 1.4kB  README.md                                   
npm notice 389B   dist/arrow.d.ts                             
npm notice 4.1kB  dist/arrow.js                               
npm notice 403B   dist/embedding/embedding_function.d.ts      
npm notice 671B   dist/embedding/embedding_function.js        
npm notice 352B   dist/embedding/openai.d.ts                  
npm notice 2.5kB  dist/embedding/openai.js                    
npm notice 10.4kB dist/index.d.ts                             
npm notice 11.9kB dist/index.js                               
npm notice 11B    dist/test/embedding/openai.d.ts             
npm notice 2.8kB  dist/test/embedding/openai.js               
npm notice 11B    dist/test/io.d.ts                           
npm notice 3.9kB  dist/test/io.js                             
npm notice 11B    dist/test/test.d.ts                         
npm notice 15.5kB dist/test/test.js                           
npm notice 1.4kB  examples/js-openai/index.js                 
npm notice 317B   examples/js-openai/package.json             
npm notice 2.3kB  examples/js-transformers/index.js           
npm notice 381B   examples/js-transformers/package.json       
npm notice 4.1kB  examples/js-youtube-transcripts/index.js    
npm notice 317B   examples/js-youtube-transcripts/package.json
npm notice 1.1kB  examples/js/index.js                        
npm notice 286B   examples/js/package.json                    
npm notice 439B   examples/ts/package.json                    
npm notice 1.1kB  examples/ts/src/index.ts                    
npm notice 179B   examples/ts/tsconfig.json                   
npm notice 1.3kB  native.js                                   
npm notice 2.5kB  package.json                                
npm notice 2.9kB  src/arrow.ts                                
npm notice 981B   src/embedding/embedding_function.ts         
npm notice 1.7kB  src/embedding/openai.ts                     
npm notice 15.9kB src/index.ts                                
npm notice 1.7kB  src/test/embedding/openai.ts                
npm notice 2.7kB  src/test/io.ts                              
npm notice 11.2kB src/test/test.ts                            
npm notice 179B   tsconfig.json                               
npm notice === Tarball Details === 
npm notice name:          vectordb                                
npm notice version:       0.1.14                                  
npm notice filename:      vectordb-0.1.14.tgz                     
npm notice package size:  20.2 kB                                 
npm notice unpacked size: 108.7 kB                                
npm notice shasum:        f642920a24b3bbccb2c22fc89a07cdc57b810310
npm notice integrity:     sha512-1tndvlDH0Nkq1[...]fcKU6dwnJRLGw==
npm notice total files:   38                                      
npm notice 
npm ERR! code ENEEDAUTH
npm ERR! need auth This command requires you to be logged in to https://registry.npmjs.org/
npm ERR! need auth You need to authorize this machine using `npm adduser`

npm ERR! A complete log of this run can be found in: /home/runner/.npm/_logs/2023-07-17T21_05_07_128Z-debug-0.log
```

--- a/.github/workflows/npm-publish.yml
+++ b/.github/workflows/npm-publish.yml
@@ -161,6 +161,7 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: 20
+          registry-url: 'https://registry.npmjs.org'
       - name: Publish to NPM
         env:
           NODE_AUTH_TOKEN: ${{ secrets.LANCEDB_NPM_REGISTRY_TOKEN }}

